### ISSUE TITLE
feat: `vm.depth()` cheatcode to return the depth of the current invariant run

### ISSUE THREAD

**Issue by mds1** [foundry-rs/foundry#2985]
### Component

Forge

### Describe the feature you would like

Two enables two use cases that I can't think of good workarounds for:
1. Logging all sequences for all runs to a file with `vm.writeLine()`
2. Skipping checks on `depth=0`, i.e. skip the "invariant must hold at the start" check. The use case here is testing that a variable monotonically increases and before any calls are made, since at the start "last value" and "current value" are the same.

**Comment by gakonst** [comment]
@joshieDo PTAL

**Comment by lucas-manuel** [comment]
Agreed this would be a great feature to add. Was thinking that it could also be leveraged for `afterAll` hook style functionality but on second thought I think it would be more efficient to just implement a new feature for this. What do you guys think?

**Comment by mds1** [comment]
Ah interesting, one downside of that is you have to hardcode the max depth since you currently can't access it (easily) from within a test, so I do agree a separate feature is probably cleaner

**Comment by lucas-manuel** [comment]
Is there a ticket for an `afterAll` hook?

**Comment by jenpaff** [comment]
the new impl of coverage guided fuzzing writes corpus and related metadata (new coverage, how many time mutated) on disk, so this info is available, marking this as not planned for now
